cairo/ethereum/cancun/trie.cairo

```diff
@@ -134,45 +207,87 @@ from ethereum.cancun.transactions import LegacyTransaction
 //     // return trie._data.get(key, trie.default)
 // }
 
-// func common_prefix_length(a: Sequence, b: Sequence) -> felt {
-//     // Implementation:
-//     // for i in range(len(a)):
-//     // if i >= len(b) or a[i] != b[i]:
-//     // return i
-//         // if i >= len(b) or a[i] != b[i]:
-//         // return i
-//             // return i
-//     // return len(a)
-// }
+func common_prefix_length(a: Bytes, b: Bytes) -> felt {
+    alloc_locals;
+    local result;
 
-// func nibble_list_to_compact(x: Bytes, is_leaf: bool) -> Bytes {
-//     // Implementation:
-//     // compact = bytearray()
-//     // if len(x) % 2 == 0:
-//     // compact.append(16 * (2 * is_leaf))
-//     // for i in range(0, len(x), 2):
-//     // compact.append(16 * x[i] + x[i + 1])
-//     // else:
-//     // compact.append(16 * (2 * is_leaf + 1) + x[0])
-//     // for i in range(1, len(x), 2):
-//     // compact.append(16 * x[i] + x[i + 1])
-//         // compact.append(16 * (2 * is_leaf))
-//         // for i in range(0, len(x), 2):
-//         // compact.append(16 * x[i] + x[i + 1])
-//             // compact.append(16 * x[i] + x[i + 1])
-//     // else:
-//         // compact.append(16 * (2 * is_leaf + 1) + x[0])
-//         // for i in range(1, len(x), 2):
-//         // compact.append(16 * x[i] + x[i + 1])
-//             // compact.append(16 * x[i] + x[i + 1])
-//     // return Bytes(compact)
-// }
+    %{ memory[fp] = oracle(ids, reference="ethereum.cancun.trie.common_prefix_length") %}
+
+    jmp common if result != 0;
+    jmp diff;
+
+    common:
+    let index = [ap - 1];
+    with_attr error_message("common_prefix_length") {
+        assert a.value.data[index - 1] = b.value.data[index - 1];
+    }
+    tempvar index = index - 1;
+    jmp common if index != 0;
+
+    diff:
+    let result = [fp];
+    if (a.value.len == result) {
+        return result;
+    }
+
+    if (b.value.len == result) {
+        return result;
+    }
+
+    with_attr error_message("common_prefix_length") {
+        assert_not_zero(a.value.data[result] - b.value.data[result]);
+    }
+
+    return result;
+}
+
+func nibble_list_to_compact(x: Bytes, is_leaf: bool) -> Bytes {
+    alloc_locals;
+    let (local compact) = alloc();
+
+    if (x.value.len == 0) {
+        assert [compact] = 16 * (2 * is_leaf.value);
+        tempvar result = Bytes(new BytesStruct(compact, 1));
+        return result;
+    }
+
+    local remainder = nondet %{ ids.x.value.len % 2 %};
+    assert remainder * (1 - remainder) = 0;
+    assert [compact] = 16 * (2 * is_leaf.value + remainder) + x.value.data[0] * remainder;
+
+    if (x.value.len == remainder) {
```

Comment:
```suggestion
    if (x.value.len == 1) {
```

I would hardcode 1 here because the zero case has been handled explicitly before